modal/_utils/async_utils.py

```diff
@@ -561,35 +561,82 @@ class StopSentinelType:
 STOP_SENTINEL = StopSentinelType()
 
 
-async def async_merge(*inputs: Union[AsyncIterable[T], Iterable[T]]) -> AsyncGenerator[T, None]:
-    queue: asyncio.Queue[Tuple[int, Union[ValueWrapper[T], ExceptionWrapper, StopSentinelType]]] = asyncio.Queue()
+async def async_merge(*iterables: Union[AsyncIterable[T], Iterable[T]]) -> AsyncGenerator[T, None]:
```

Comment:
A bit of a late realization, but I just realized it might be nice to enforce the signature of these to only take async iterables, and let the user of the function apply `sync_or_async_iter` on the input iterators when there is ambiguity. Makes it a bit more explicit in usage

Reply:
makes sense, do we need them to be async iterables though? Updated the branch https://github.com/modal-labs/modal-client/pull/2421 this is taken from to just be async generators since that is what we return from `sync_or_async_iter` anyway